CMakeLists.txt

```diff
@@ -54,17 +54,19 @@ SEARCH_FOR_BOOST()
 # Libtool dynamic loading
 # This project does not use Libtool directly but still uses ltdl for
 # plug-in loading.
-SET(LIB_LTDL LIB_LTDL-NOTFOUND)
-FIND_LIBRARY(LIB_LTDL libltdl${CMAKE_SHARED_LIBRARY_SUFFIX})
-IF (NOT LIB_LTDL)
-  MESSAGE(FATAL_ERROR
-    "Failed to find Libtool ltdl library, check that Libtool ltdl is installed.")
-ENDIF()
 INCLUDE(CheckIncludeFileCXX)
-CHECK_INCLUDE_FILE_CXX(ltdl.h LTDL_H_FOUND)
-IF (NOT LTDL_H_FOUND)
-  MESSAGE(FATAL_ERROR
-    "Failed to find ltdl.h, check that Libtool ltdl is installed.")
+IF(NOT WIN32)
+  SET(LIB_LTDL LIB_LTDL-NOTFOUND)
+  FIND_LIBRARY(LIB_LTDL libltdl${CMAKE_SHARED_LIBRARY_SUFFIX})
+  IF (NOT LIB_LTDL)
+    MESSAGE(FATAL_ERROR
+      "Failed to find Libtool ltdl library, check that Libtool ltdl is installed.")
+  ENDIF()
+  CHECK_INCLUDE_FILE_CXX(ltdl.h LTDL_H_FOUND)
+  IF (NOT LTDL_H_FOUND)
+    MESSAGE(FATAL_ERROR
+      "Failed to find ltdl.h, check that Libtool ltdl is installed.")
+  ENDIF()
 ENDIF()
```

Comment:
With [ltld-win32](https://github.com/gergondet/ltdl-win32), we can use [this](https://github.com/roboptim/roboptim-core/blob/95c24e529d2aebea875f91380809c0e4904769cd/CMakeLists.txt#L214-L231), i.e. pkg-config check on Windows, and fallback to the current check for other platforms.